inc/assembly_report/assembly_report.hpp

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright 2014-2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef ASSEMBLY_REPORT_HPP
+#define ASSEMBLY_REPORT_HPP
+
+#include <map>
+#include <set>
+#include <string>
+#include <vector>
+
+#include <boost/algorithm/string/trim.hpp>
+#include "util/stream_utils.hpp"
+#include "util/string_utils.hpp"
+
+namespace ebi
+{
+  namespace assembly_report
+  {
+    /*
+     * Struct to contain all the synonyms of a contig
+     */
+    struct ContigSynonyms {
+        std::vector<std::string> synonyms;
+
+        ContigSynonyms() {
+
+        }
+
+        ContigSynonyms(std::string contig) {
+            synonyms.push_back(contig);
+        }
+
+        ContigSynonyms(std::vector<std::string> & synonyms) : synonyms(synonyms) {
+
+        }
+    };
+
+    /*
+     * Map to map all contigs to their synonym_maps
+     */
+    class SynonymsMap {
+        std::map<std::string,int> index_map;
+        std::vector<ContigSynonyms> contigs;
+        /*
+         * set of names that needs to be ignored if present in assembly report
+         */
+        std::set<std::string> ignore_contig{"na"};
+
+        ContigSynonyms extract_synonyms(std::vector<std::string> & synonyms) {
+            std::vector<int> synonym_indices{0,4,6,9};
+            ContigSynonyms contig_synonyms;
+            for (auto index : synonym_indices) {
+                if(ignore_contig.find(synonyms[index]) != ignore_contig.end()) {
+                    continue;
+                }
+                contig_synonyms.synonyms.push_back(synonyms[index]);
+            }
+
+            return contig_synonyms;
+        }
+
+        public:
+        bool is_contig_available(std::string contig) {
```

Comment:
I kept in this way in case we need to log some error in case of failure or something else that we can do in case of false condtion. the structure I wrote was having more flexibility of adding some complementry code into it. still I would be fine for changing it to one line. as if required in future it may be done again. but still logically it wont make any difference. compiler does it for us it will convert it into one line.

Reply:
will covert it into one line